examples/language-modeling/run_clm.py

```diff
@@ -390,17 +391,19 @@ def main():
             if model_args.torch_dtype in ["auto", None]
             else getattr(torch, model_args.torch_dtype)
         )
-        model = AutoModelForCausalLM.from_pretrained(
-            model_args.model_name_or_path,
-            from_tf=bool(".ckpt" in model_args.model_name_or_path),
-            config=config,
-            cache_dir=model_args.cache_dir,
-            revision=model_args.model_revision,
-            use_auth_token=True if model_args.use_auth_token else None,
-            torch_dtype=torch_dtype,
-        )
+        with lazy_load_for_parallelism(tensor_parallel_size=training_args.tensor_parallel_size):
+            model = AutoModelForCausalLM.from_pretrained(
```

Comment:
Also, couldn't you take advantage of the `low_cpu_mem_usage` parameter ?

Reply:
`low_cpu_mem_usage` loads ~1x model size in RAM. We actually cannot afford that at all here.

Let's the example is run on 32 cores, if the model is 7B parameters big, it's 14 GB in ram for one worker for nothing since it's going to be parallelized later on.

Instead, we skip loading altogether and postpone it to when we parallelize the model.